### PR TITLE
added real peak values at each breath

### PR DESCRIPTION
**Pull request by JonBruchim**
Signed-off-by: rinslow <yonbruchim@gmail.com>

### CODE REVIEW

File: algo.py [hunk 1]
```diff
@@ -33,21 +35,35 @@ def _handle_intake(self, flow, pressure):
         self._currently_breathed_volume += \
             (flow * sampling_interval_in_minutes)
 
-        if self._currently_breathed_volume > self._data_store.flow_max_threshold.value:
+        if self._currently_breathed_volume >\
```
Comment (avrahamshukron) [inline algo.py]: `currently_breathed_volume` is long and complicated name.
Lets rename to `current_intake_volume`

File: algo.py [hunk 2]
```diff
@@ -33,21 +35,35 @@ def _handle_intake(self, flow, pressure):
...
             self.breathing_alert = AlertCodes.BREATHING_VOLUME_HIGH
 
         if pressure <= self._data_store.breathing_threshold:
             self._has_crossed_first_cycle = True
 
+        if self._intake_max_pressure < pressure:
+            self._intake_max_pressure = pressure
+
+        if self._intake_max_flow < flow:
+            self._intake_max_flow = flow
```
Comment (avrahamshukron) [inline algo.py]: Could be one-liner:

```suggestion
        self._intake_max_pressure = max(pressure, self._intake_max_pressure):
        self._intake_max_flow = max(flow, self._intake_max_flow)
```

File: algo.py [hunk 3]
```diff
@@ -33,21 +35,35 @@ def _handle_intake(self, flow, pressure):
...
         """We are not giving patient air anymore."""
 
-        if self._currently_breathed_volume < self._data_store.flow_min_threshold.value and \
+        if self._currently_breathed_volume <\
+           self._data_store.flow_min_threshold.value and \
                 self._has_crossed_first_cycle:
 
             self.breathing_alert = AlertCodes.BREATHING_VOLUME_LOW
 
+        self._data_store.set_intake_peaks(self._intake_max_pressure, self._intake_max_pressure,
```
Comment (avrahamshukron) [inline algo.py]: 1. Why do we need it in the data store?
2. Magic Number. This should be a constant somewhere

File: data/data_store.py [hunk 4]
```diff
@@ -47,6 +47,8 @@ def __init__(self):
         self.x_axis = range(0, self.samples_in_graph_amount)
 
         self.volume = 0
+        self.intake_peak_flow = 0
+        self.intake_peak_pressure = 0
```
Comment (avrahamshukron) [inline data/data_store.py]: Why is it in `DataStore`?

Comment (JonBruchim) [reply data/data_store.py]: graph_summaries uses both of these parameters to update
maximum peak values.
We could use our publish-subscribe module.
these parameters have only one costumer.

File: algo.py (hunk 3 above)
Comment (JonBruchim) [reply algo.py]: 1. same as last comment about graph_summary
2. will do soon

File: algo.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
 class Sampler(threading.Thread):
     SAMPLING_INTERVAL = 0.02  # sec
     MS_IN_MIN = 60 * 1000
+    VOLUME_CALIBRATION_MULTIPLYER = 100000
```
Comment (Reznic) [inline algo.py]: This should be only for the mock (sensor simulation mode).
If we are connected to real flow sensor, this fucks the sample values

Comment (JonBruchim) [reply algo.py]: Want to remove it from the code?
because mock related calculations are in drivers/mocks
and volume calculation is in sampler

**Pull request merged**